Java/core/src/main/java/com/amazon/randomcutforest/preprocessor/transform/SubtractMATransformer.java

```diff
@@ -32,4 +32,18 @@ public SubtractMATransformer(double[] weights, Deviation[] deviations) {
     protected double getShift(int i, Deviation[] devs) {
         return devs[i].getMean();
     }
+
+    @Override
+    public void invertForecastRange(RangeVector ranges, int baseDimension, double[] previousInput,
+                                    double[] correction) {
+        super.invertForecastRange(ranges,baseDimension,previousInput,correction);
+        int horizon = ranges.values.length / baseDimension;
+        int inputLength = weights.length;
+        for (int i = 0; i < horizon; i++) {
+            for (int j = 0; j < inputLength; j++) {
+                ranges.shift(i * baseDimension + j,
+                        (float) (0.5 * getDrift(j, deviations)));
```

Comment:
Why do we need to add half of the drift (which I understand is the mean of step difference) to all values in the horizon?

Reply:
This seems to be the gap left over from the general case even for a simple linear case -- because we updated the moving average (before calling this code) and some of the step difference then was already added. Note that this corresponds to what was shifted. But SUBTRACT_MA seems to not be a very stable transformation -- added  that to TransformTest  and then took it away (because the test was too non-deterministic), see @EnumSource(value = TransformMethod.class, names = {"NORMALIZE", "NORMALIZE_DIFFERENCE",  "DIFFERENCE"}).

I'd recommend using NORMALIZE or NORMALIZE_DIFFERENCE -- but added this just so that we have captured correction.

Reply:
I think it makes sense to drop this change and just keep the discussion that Normalize/Normalize_difference are the recommended ones and being tested more.